src/OpenTelemetry.AutoInstrumentation/Loading/Initializers/GraphQLInitializer.cs

```diff
@@ -0,0 +1,54 @@
+// <copyright file="GraphQLInitializer.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Reflection;
+using OpenTelemetry.AutoInstrumentation.Configurations;
+using OpenTelemetry.AutoInstrumentation.Plugins;
+
+namespace OpenTelemetry.AutoInstrumentation.Loading.Initializers;
+
+internal class GraphQLInitializer : InstrumentationInitializer
+{
+    private readonly PluginManager _pluginManager;
+    private readonly TracerSettings _tracerSettings;
+
+    public GraphQLInitializer(PluginManager pluginManager, TracerSettings tracerSettings)
+    : base("GraphQL")
+    {
+        _pluginManager = pluginManager;
+        _tracerSettings = tracerSettings;
+    }
+
+    public override void Initialize(ILifespanManager lifespanManager)
+    {
+        var initializerType = Type.GetType("OpenTelemetry.AutoInstrumentation.Initializer, GraphQL");
+        if (initializerType == null)
+        {
+            return;
+        }
+
+        var optionsType = Type.GetType("GraphQL.Telemetry.GraphQLTelemetryOptions, GraphQL")!;
+        var optionsInstance = Activator.CreateInstance(optionsType)!;
```

Comment:
The the same vein, I would have expected this to by a dynamic mapping of options -- or, to not support options at all, just as if the library emitted tracing events by default.

Reply:
This should become better when OTel config file is finally speced. Currently introducing new envs is also on hold.
But in the end environment controller (end user) and OTel vendor (via plugins) must have control over options. Some of the options can control sending sensitive data over the wire, which is extremely important to control.